namedquery/src/main/java/software/amazon/athena/namedquery/HandlerUtils.java

```diff
@@ -0,0 +1,33 @@
+package software.amazon.athena.namedquery;
+
+import software.amazon.awssdk.services.athena.model.AthenaException;
+import software.amazon.awssdk.services.athena.model.InternalServerException;
+import software.amazon.awssdk.services.athena.model.InvalidRequestException;
+import software.amazon.cloudformation.exceptions.CfnGeneralServiceException;
+import software.amazon.cloudformation.exceptions.CfnInvalidRequestException;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+
+import static software.amazon.athena.namedquery.ResourceModel.TYPE_NAME;
+
+final class HandlerUtils {
+
+  private HandlerUtils() {
+    // Intentionally left blank
+  }
+
+  private static final String NAMEDQUERY_NOT_FOUND = "does not exist";
+
+  static RuntimeException translateAthenaException(AthenaException e, String resourceIdentifier) {
+    if (e instanceof InvalidRequestException) {
+      if (e.getMessage() != null && e.getMessage().contains(NAMEDQUERY_NOT_FOUND)) {
```

Comment:
Will it be used for other resources, ex, workgroup. Maybe we could generalized it.

Reply:
I looked into creating a common module with overlapping logic between the different resources, but it didn't seem like a simple change. I wanted to get these fixes out quickly as there is a campaign from CFN to fix these tests. I'll look back into refactoring it a bit once we have the tests passing and there is less urgency

Reply:
Sure. We could change it later.